File: Angular Client/models/followup.ts
import {FormGroup} from '@angular/forms';

export interface Followup {
  followUpId: number;
  animalId: number;
  date: Date;
  cause: string;
  status: boolean;
}

export class FollowupEntity implements Followup{
  constructor() {
    this.followUpId = 0;
    this.animalId = 0;
    this.date = new Date();
    this.cause = '';
    this.status = false;
  }
  followUpId: number;
  animalId: number;
  date: Date;
  cause: string;
  status: boolean;

  setData(form: FormGroup): void{
    this.followUpId = form.get('followUpId')?.value;
    this.animalId = form.get('animalId')?.value;
    this.date = form.get('date')?.value;
    this.cause = form.get('cause')?.value;
    this.status = form.get('status')?.value;
  }
}

File: ASP.NET Core Backend/Petadmin.Core/Models/FollowUp.cs
using System;
using PetAdmin.Core.Interfaces;

namespace Petadmin.Core.Models
{
    public class FollowUp : IGenericDbEntity
    {
        /// <summary>
        /// Keeps the database id of the FollowUp.
        /// </summary>
        public int FollowUpId { get; set; }

        /// <summary>
        /// Keeps the database id of the animal.
        /// </summary>
        public int AnimalId { get; set; }

        /// <summary>
        /// Gets or sets the date of FollowUp.
        /// </summary>
        public DateTime? Date { get; set; }
        
        /// <summary>
        /// Gets or sets the Cause of the FollowUp.
        /// </summary>
        public string Cause { get; set; }

        /// <summary>
        /// Gets or sets the Status of the FollowUp.
        /// </summary>
        public bool Status { get; set; }
    }
}
